torch/csrc/jit/script/class_type.cpp

```diff
@@ -5,8 +5,8 @@ namespace c10 {
 
 // This file exists because we need to reference module.h, which we can't from
 // c10. Sigh...
-Function* ClassType::getMethod(const std::string& name) const {
-  return compilation_unit_->find_function(name).get();
+std::shared_ptr<Function> ClassType::getMethod(const std::string& name) const {
```

Comment:
I see a lot of places where we're turning Function refs/ptrs into owning pointers. What's the motivation? Seems like the ownership hasn't really changed since a CompilationUnit still owns the actual functions.

Reply:
We are moving to shared ownership of Function objects across compilation units (and other things). Now that you _can_ own a script::Function, I've been updating the ownership incrementally to reflect that. We might need more tweaks once we finish the 'correct' ownership model.